### PR TITLE
PARQUET-932: Add option to build parquet library with minimal dependency

### PR DESCRIPTION
**Pull request by majetideepak**
If users want to build only the parquet library, thirdparty headers are sufficient and GTEST and GBENCHMARK are not required.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -103,10 +103,14 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   option(PARQUET_RPATH_ORIGIN
     "Build Parquet libraries with RATH set to \$ORIGIN"
     OFF)
+  option(PARQUET_MINIMAL_DEPENDENCY
+    "Depend only on Thirdparty headers to build libparquet. Always OFF if building binaries"
+    OFF)
```
Comment (majetideepak) [inline CMakeLists.txt]: We could also implicitly set this value to ``OFF`` by default and set it to `ON` if `PARQUET_BUILD_TESTS` and `PARQUET_BUILD_EXECUTABLES` and `PARQUET_BUILD_BENCHMARKS` are all `OFF`

**Comment by majetideepak** [comment]
@xhochy, @wesm  Can you give your feedback on this? I would like this patch to be committed soon.Thanks!

File: cmake_modules/FindArrow.cmake [hunk 2]
```diff
@@ -51,8 +51,10 @@ find_library(ARROW_IO_LIB_PATH NAMES arrow_io
   ${ARROW_SEARCH_LIB_PATH}
   NO_DEFAULT_PATH)
 
-if (ARROW_INCLUDE_DIR AND ARROW_LIB_PATH)
+if (ARROW_INCLUDE_DIR AND (PARQUET_MINIMAL_DEPENDENCY OR ARROW_LIB_PATH))
   set(ARROW_FOUND TRUE)
+  set(ARROW_HEADER_NAME arrow/arrow.h)
```
Comment (wesm) [inline cmake_modules/FindArrow.cmake]: It's called `arrow/api.h`

File: cmake_modules/FindArrow.cmake [hunk 3]
```diff
@@ -51,8 +51,10 @@ find_library(ARROW_IO_LIB_PATH NAMES arrow_io
   ${ARROW_SEARCH_LIB_PATH}
   NO_DEFAULT_PATH)
 
-if (ARROW_INCLUDE_DIR AND ARROW_LIB_PATH)
+if (ARROW_INCLUDE_DIR AND (PARQUET_MINIMAL_DEPENDENCY OR ARROW_LIB_PATH))
```
Comment (wesm) [inline cmake_modules/FindArrow.cmake]: Just so I understand, you have a build environment where only the headers are available, and no libraries?

Comment (majetideepak) [inline cmake_modules/FindArrow.cmake]: That is correct.

File: cmake_modules/FindArrow.cmake (hunk 2 above)
Comment (majetideepak) [inline cmake_modules/FindArrow.cmake]: Fixed.

**Comment by wesm** [comment]
+1, merging this

**Pull request closed without merging**